evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -194,3 +200,27 @@ def transform(self, X, y=None):
             y_imputed = ww.init_series(y_imputed)
 
         return X_not_all_null, y_imputed
+
+    def _handle_nullable_types(self, X=None, y=None):
+        """Transforms X and y to remove any incompatible nullable types for the time series imputer when the interpolate method is used.
+
+        Args:
+            X (pd.DataFrame, optional): Input data to a component of shape [n_samples, n_features].
+                May contain nullable types.
+            y (pd.Series, optional): The target of length [n_samples]. May contain nullable types.
+
+        Returns:
+            X, y with any incompatible nullable types downcasted to compatible equivalents when interpolate is used. Is NoOp otherwise.
+        """
+        if self._impute_target == "interpolate":
+            # For BooleanNullable, we have to avoid Categorical columns
+            # since the category dtype also has incompatibilities with linear interpolate, which is expected
+            # --> i think this is essentially what happens now but won't that make floating point values that cant be turned back into bools?
+            if isinstance(y.ww.logical_type, BooleanNullable):
+                y = ww.init_series(y, Double)
```

Comment:
Noticed that `interpolate()` cannot handle `category` dtypes either. But this, I think, is by design, as the default interpolation is "linear" which doesnt make sense for non numeric values. This is problematic if we try to turn BooleanNullable into Categorical ltype when nans are present, so I'm intercepting it and converting to `Double`, which essentially the current behavior.

In the long run, though, I think we don't want to be doing linear interpolation on boolean columns ever - we don't want to end up with `1.5` in a column of all 1s and 0s. I'll open a separate evalml ticket for handling that, though.

Reply:
makes sense!

Reply:
We already don't allow using interpolation with categorical or boolean types. From the docstring:
```
        categorical_impute_strategy (string): Impute strategy to use for string, object, boolean, categorical dtypes.
            Valid values include "backwards_fill" and "forwards_fill". Defaults to "forwards_fill".
        numeric_impute_strategy (string): Impute strategy to use for numeric columns. Valid values include
            "backwards_fill", "forwards_fill", and "interpolate". Defaults to "interpolate".
```
It's enforced in a couple different places, between checking the arguments and how we select splitting between numeric and categorical!

Reply:
Ah, very true - good point!